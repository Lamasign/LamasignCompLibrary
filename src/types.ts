import {
	AccordionVariantsArray,
	AccordionVariantsEnum,
	AccordionVariantsMap,
} from "./components/accordion/BAccordion.types"


import {
	BadgeColorsArray,
	BadgeColorsEnum,
	BadgeColorsMap,
	BadgeSizeEnum,
	BadgeSizesArray,
	BadgeSizesMap,
	BadgeVariantsArray,
	BadgeVariantsEnum,
	BadgeVariantsMap,
} from './components/Badge/BBadge.types'

import {
	ButtonSizesEnum,
	ButtonVariantsEnum,
	ButtonTagsEnum,
	ButtonTypesEnum,
	ButtonSizesArray,
	ButtonSizesMap,
	ButtonVariantsArray,
	ButtonVariantsMap,
	ButtonTagsArray,
	ButtonTypesArray,
} from './components/Button/BButton.types'

import {
	CheckboxVariantsEnum,
	CheckboxVariantsArray,
	CheckboxsVariantsMap,
} from './components/Checkbox/BCheckbox.types'

import {
	DropdownVariantsEnum,
	DropdownDirectionEnum,
	DropdownDirectionMap,
	DropdownVariantsArray,
	DropdownVariantsMap,
} from './components/Dropdown/BDropdown.types'

import {
	FIELD_STATUS_INJECTION_KEY,
	FieldStatusesArray,
	FieldStatusesEnum,
	FieldStatusesClassesMap,
	FieldStatusesTypes,
	FieldMessageStatusesClassesMap
} from './components/Field/BField.types'

import {
	InputFileStatusesArray,
	InputFileStatusesClassesMap,
	InputFileStatusesEnum
} from './components/FileInput/BFileInput.types'

import {
	InputTypesEnum,
	InputTypesArray,
	InputStatusesArray,
	InputStatusesClassesMap,
	InputStatusesEnum,
	InputStatusesTypes,
	InputTypes,
} from './components/Input/BInput.types'

import {
	LinkDisabledClassesMap,
	LinkTags,
	LinkTagsArray,
	LinkTagsEnum,
	LinkVariantsArray,
	LinkVariantsEnum,
	LinkVariantsMap,
} from './components/Link/BLink.types'

import {
	MessageVariantsArray,
	MessageVariantsEnum,
	MessageVariantsMap,
} from './components/Message/BMessage.types'

import {
	NumberInputStatusesArray,
	NumberInputStatusesClassesMap,
	NumberInputStatusesEnum,
	NumberInputStatusesTypes,
} from './components/NumberInput/BNumberInput.types'

import {
	SelectColorsEnum,
	SelectColorsArray,
	SelectColorsMap,
} from './components/Select/BSelect.types'

import {
	SkeletonVariantsEnum,
	SkeletonVariantsArray,
	SkeletonVariantsMap,
	SkeletonAnimationsArray,
	SkeletonAnimationsEnum,
	SkeletonAnimationsMap,
	SkeletonColorsArray,
	SkeletonColorsEnum,
	SkeletonColorsMap,
} from './components/Skeleton/BSkeleton.types'

import {
	SwitchVariantsArray,
	SwitchVariantsEnum,
	SwitchVariantsMap,
} from './components/Switch/BSwitch.types'

import {
	TableColumnHeaderVariantsMap,
	TableColumnVariantsArray,
	TableColumnVariantsEnum,
	TableColumnVariantsMap,
} from './components/Table/BTable.types'

import {
	TagVariantsArray,
	TagVariantsEnum,
	TagVariantsMap,
	TagsVariantsMap,
} from './components/Tag/BTag.types'

import {
	ToastVariantsArray,
	ToastVariantsMap,
	ToastVariantsEnum,
} from './components/Toast/BToast.types'

import {
	TooltipPositionEnum,
	TooltipPositionsArray,
	TooltipPositionsMap,
	TooltipVariantsArray,
	TooltipVariantsEnum,
	TooltipVariantsMap,
} from './components/Tooltip/BTooltip.types'

export {
	AccordionVariantsArray,
	AccordionVariantsEnum,
	AccordionVariantsMap,
	BadgeColorsArray,
	BadgeColorsEnum,
	BadgeColorsMap,
	BadgeSizeEnum,
	BadgeSizesArray,
	BadgeSizesMap,
	BadgeVariantsArray,
	BadgeVariantsEnum,
	BadgeVariantsMap,
	ButtonSizesEnum,
	ButtonVariantsEnum,
	ButtonTagsEnum,
	ButtonTypesEnum,
	ButtonSizesArray,
	ButtonSizesMap,
	ButtonVariantsArray,
	ButtonVariantsMap,
	ButtonTagsArray,
	ButtonTypesArray,
	CheckboxVariantsEnum,
	CheckboxVariantsArray,
	CheckboxsVariantsMap,
	DropdownVariantsEnum,
	DropdownDirectionEnum,
	DropdownDirectionMap,
	DropdownVariantsArray,
	DropdownVariantsMap,
	FIELD_STATUS_INJECTION_KEY,
	FieldStatusesArray,
	FieldStatusesEnum,
	FieldStatusesClassesMap,
	FieldStatusesTypes,
	FieldMessageStatusesClassesMap,
	InputFileStatusesArray,
	InputFileStatusesClassesMap,
	InputFileStatusesEnum,
	InputTypesEnum,
	InputTypesArray,
	InputStatusesArray,
	InputStatusesClassesMap,
	InputStatusesEnum,
	InputStatusesTypes,
	InputTypes,
	LinkDisabledClassesMap,
	LinkTags,
	LinkTagsArray,
	LinkTagsEnum,
	LinkVariantsArray,
	LinkVariantsEnum,
	LinkVariantsMap,
	MessageVariantsArray,
	MessageVariantsEnum,
	MessageVariantsMap,
	NumberInputStatusesArray,
	NumberInputStatusesClassesMap,
	NumberInputStatusesEnum,
	NumberInputStatusesTypes,
	SelectColorsEnum,
	SelectColorsArray,
	SelectColorsMap,
	SkeletonVariantsEnum,
	SkeletonVariantsArray,
	SkeletonVariantsMap,
	SkeletonAnimationsArray,
	SkeletonAnimationsEnum,
	SkeletonAnimationsMap,
	SkeletonColorsArray,
	SkeletonColorsEnum,
	SkeletonColorsMap,
	SwitchVariantsArray,
	SwitchVariantsEnum,
	SwitchVariantsMap,
	TableColumnHeaderVariantsMap,
	TableColumnVariantsArray,
	TableColumnVariantsEnum,
	TableColumnVariantsMap,
	TagVariantsArray,
	TagVariantsEnum,
	TagVariantsMap,
	TagsVariantsMap,
	ToastVariantsArray,
	ToastVariantsMap,
	ToastVariantsEnum,
	TooltipPositionEnum,
	TooltipPositionsArray,
	TooltipPositionsMap,
	TooltipVariantsArray,
	TooltipVariantsEnum,
	TooltipVariantsMap,
}
