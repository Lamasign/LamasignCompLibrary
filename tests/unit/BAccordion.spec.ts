import { AccordionVariantsArray } from './../../src/components/accordion/BAccordion.types';
import BAccordion from '@/components/accordion/BAccordion.vue'
import { createWrapper } from './utils'

describe('BAccordion.vue', () => {
	it('renders the content of his default slot', () => {
		expect.assertions(1)
		const slotContent = "content slot"
		const wrapper = createWrapper(BAccordion, {
			slots: {
				default: slotContent
			}
		})
		expect(wrapper.text()).toMatch(slotContent)
	})

	it('renders the right classes based on prop called "variant"', () => {
		expect.assertions(AccordionVariantsArray.length + 1)
		const defaultWrapper = createWrapper(BAccordion)
		expect(defaultWrapper.element).toMatchSnapshot()
		AccordionVariantsArray.forEach(variant => {
			const wrapperWithProps = createWrapper(BAccordion, {
				propsData: {
					variant,
				}
			})
			expect(wrapperWithProps.element).toMatchSnapshot()
		})
	})

	it('emit click "event open" when Accordion title is clicked', async () => {
		const wrapper = createWrapper(BAccordion)
		expect(wrapper.classes('d-block')).toBe(false)
		await wrapper.find('button').trigger('click')
		const open = wrapper.emitted('open')
		expect(open).toBeTruthy()
	})
})