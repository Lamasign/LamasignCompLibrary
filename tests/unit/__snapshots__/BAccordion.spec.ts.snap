// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BAccordion.vue renders the right classes based on prop called "variant" 1`] = `
<div
  class="border-2 p-2 rounded mt-2 bg-white text-dark hover:bg-blue-dark rounded cursor-pointer"
>
  <button>
    
  </button>
   
  <div
    class="mt-4 border-2 p-2 rounded hidden"
  />
</div>
`;

exports[`BAccordion.vue renders the right classes based on prop called "variant" 2`] = `
<div
  class="border-2 p-2 rounded mt-2 bg-white text-dark hover:bg-blue-dark rounded cursor-pointer"
>
  <button>
    
  </button>
   
  <div
    class="mt-4 border-2 p-2 rounded hidden"
  />
</div>
`;

exports[`BAccordion.vue renders the right classes based on prop called "variant" 3`] = `
<div
  class="border-2 p-2 rounded mt-2 bg-orange text-white hover:bg-orange-dark rounded cursor-pointer"
>
  <button>
    
  </button>
   
  <div
    class="mt-4 border-2 p-2 rounded hidden"
  />
</div>
`;

exports[`BAccordion.vue renders the right classes based on prop called "variant" 4`] = `
<div
  class="border-2 p-2 rounded mt-2 bg-red text-white hover:bg-red-dark rounded cursor-pointer"
>
  <button>
    
  </button>
   
  <div
    class="mt-4 border-2 p-2 rounded hidden"
  />
</div>
`;

exports[`BAccordion.vue renders the right classes based on prop called "variant" 5`] = `
<div
  class="border-2 p-2 rounded mt-2 bg-green text-white hover:bg-green-dark rounded cursor-pointer"
>
  <button>
    
  </button>
   
  <div
    class="mt-4 border-2 p-2 rounded hidden"
  />
</div>
`;
